Markup of the homepage v2, category, product details pages

annasvrv opened the pull request.

annasvrv committed:
added markup of the homepage v2

annasvrv committed:
created markup of category.html page

annasvrv committed:
added link tag to product items

annasvrv committed:
Changed close button entity

annasvrv committed:
Created markup of product-details.html page

annasvrv committed:
Added search bar and removed login btn for consistency with homepageV2

arkhipenkav commented on index.html:
```diff
@@ -3,7 +3,478 @@
+            <a href="#" class="btn-explore">
+              <img src="/" alt="play button" />
+            </a>
+            Watch About Us
```
I believe the text should be nested inside of an `<a>` here so an entire element will be clickable, not only an icon.
After this change you will not need an extra `div` any more that you use as a wrapper.
```
<a href="#" class="btn-explore">
    <img src="/" alt="play button" />
    Watch About Us
</a>
```

annasvrv replied:
Thank you Olya for all your comments. I will go through them one by one between current tasks. 
I hope you don't mind if I leave a small comment "fixed" under your comments for myself to see where I stopped?

#fixed

arkhipenkav replied:
I hope they will be useful for you. Sure, `fixed` comment will be very appropriate here 👍

arkhipenkav commented on category.html:
```diff
@@ -0,0 +1,499 @@
+          </li>
+          <li>
+            <a href="./creator.html">Creators</a>
+            <p class="label">NEW</p>
```
`<p>` is a block element, we use it for meaningful content like a paragraph in a book. It will take an all space in a row relatively to a parent element. Otherwise, span - an inline container that is used for decorative elements like this. The best practice for label will be to use `<span>`.

annasvrv replied:
Thank you for this comment! I was wondering which tag to use but couldn't think of span. It is certainly a great example. 

#fixed

arkhipenkav commented on category.html:
```diff
@@ -0,0 +1,499 @@
+    <!-- Main -->
+    <main>
+      <!-- Title section -->
+      <section class="category-title">
```
Section will be an extra wrapper here, do we really need it?

annasvrv replied:
#fixed

arkhipenkav commented on category.html:
```diff
@@ -0,0 +1,499 @@
+
+      <aside>
+        <!-- Categories -->
+        <article class="filter-card">
```
I believe that `<section>` is a correct choice here. 

**A simple way to decide what to choose.**
Is this block independent and can be used anywhere on the website? 
- yes -> it is an article
- no -> it is a section

You can read a few articles from the course materials to use these two elements wisely
[The `<section>` Tag in HTML](https://www.scaler.com/topics/section-tag-in-html/), [The `<article>` Tag in HTML](https://www.scaler.com/topics/article-tag-in-html/): Details on the semantic use of the `<section>`, `<article>`, emphasizing their importance in structuring content and improving accessibility.

[Using Article and Section Elements for Accessibility](https://www.smashingmagazine.com/2022/07/article-section-elements-accessibility/): Recommendations for using the `<article>` and `<section>` elements to improve content accessibility, ensuring better interaction with the website for all users.

annasvrv replied:
Thank you for the tip of the way how to choose between `<article>` and `<section> `tags. 

#fixed

arkhipenkav replied:
It is great that you have used new knowledge and add a fieldset :)

arkhipenkav commented on category.html:
```diff
@@ -0,0 +1,499 @@
+
+        <div class="price-container">
+          <h2>Price</h2>
+          <div class="range-input">
```
I'm impressed 👍  You made a good self research. You will know about `input type="range"` from the lesson about forms.
Just a small remark here. This functionality possible only with JavaScript help. So for styling it is enough to have just a div with buttons here. But still, you will get acquainted with it soon.

annasvrv replied:
Thank you for your comment

#ignore comment ))

I'm a bit confused what should I change here. Do you mean I don't have to put values attributes? Could you clarify it, please. Thank you. 

`<input
              type="range"
              min="5"
              max="50"
              value="22"
              step="1"
              class="min-range"
            />
            <input
              type="range"
              min="5"
              max="50"
              value="38"
              step="1"
              class="max-range"
            />` 
change to? : 
`<input   type="range"   class="range" />
  <input   type="range"   class="range" />`

#ignore comment ))

annasvrv replied:
I have tried to  fix it according to lecture ) 

#fixed

arkhipenkav commented on category.html:
```diff
@@ -0,0 +1,499 @@
+          <!-- Dropdown menu of products -->
+          <li>
+            <a href="#" class="dropdown active">Products</a>
+            <ul class="dropdown-content">
```
Nice improvisation with a nested list here. Very good solution 👍

arkhipenkav commented on category.html:
```diff
@@ -0,0 +1,499 @@
+      <!-- Search bar -->
+      <div class="search-bar">
+        <input type="text" placeholder="Search for Digital Assets" />
+        <button>
```
You haven't had a lesson about forms yet. So for now let's just add an attribute type for a button. The reason is the tag has a default value for this attribute - `submit`. We have to control its behavior to keep the nasty bugs away.

annasvrv replied:
#fixed

arkhipenkav commented on category.html:
```diff
@@ -0,0 +1,499 @@
+
+      <!-- Login and cart buttons -->
+      <div class="buttons">
+        <!-- <button class="btn-account">
```
It is not a good practice to leave a commented code without any expanation why it is commented. I saw that you ask Liza about 2 different states of header, so I believe you can choose one - search or account button - and remove another one completely.

annasvrv replied:
#fixed

annasvrv replied:
I have removed login button and left search form. I also asked Liza about footer. She said we can choose one style as well. Since I have chosen homepage 2 design, I apply homepage2 footer design to other pages as well.

arkhipenkav commented on category.html:
```diff
@@ -0,0 +1,499 @@
+          <img src="" alt="" />
+          Profile login
+        </button> -->
+        <a href="./cart.html" class="btn-cart">
```
Nice job! A tricky one; it is a link that looks like a button and you've chosen a correct tag.

arkhipenkav commented on category.html:
```diff
@@ -0,0 +1,499 @@
+          <h2>Categories</h2>
+          <ul>
+            <li class="category-item">
+              <input type="checkbox" checked />
```
It is great that you've tried to use inputs and checkboxes! 👍 
You will have a lesson about forms soon. Let's leave here a comment with TODO. Ex. `<!-- TODO check this filters after a lesson about forms -->`
It's a very common practice not to forget to change some code after you will have a necessary information or data.

annasvrv replied:
#fixed

arkhipenkav commented on category.html:
```diff
@@ -0,0 +1,499 @@
+        </article>
+
+        <!-- Format Type -->
+        <article class="filter-card">
```
The same here, it is a `section` not an `article`. Please, check everywhere

annasvrv replied:
#fixed

arkhipenkav commented on category.html:
```diff
@@ -0,0 +1,499 @@
+          </div>
+        </div>
+
+        <button class="clearall-btn">Clear all filters</button>
```
You should add a `type` attribute for a button. Check it everywhere, please

annasvrv replied:
#fixed

arkhipenkav commented on category.html:
```diff
@@ -0,0 +1,499 @@
+
+    <!-- Footer -->
+    <footer>
+      <section class="footer-sitemap">
```
Please, rethink a classnaming here and below. `sitemap` is a bit confusing because it is a reserved name for a separate page with all links that indexing by search robots. For example https://www.apple.com/sitemap/. It is more like a `footer-navigation` or `footer-main`

annasvrv replied:
#fixed

arkhipenkav commented on category.html:
```diff
@@ -0,0 +1,499 @@
+          </p>
+        </div>
+        <!-- Sitemap section -->
+        <div class="sitemap">
```
The same here, please, choose a different class name. And why not a `section` tag? :)

annasvrv replied:
#fixed

arkhipenkav commented on category.html:
```diff
@@ -0,0 +1,499 @@
+          </div>
+          <div class="price-field">
+            <p class="min-value">$5</p>
+            <p class="max-value">$50</p>
```
```
<p class="min-value">$5</p>
<p class="max-value">$50</p>
```
These two elements look identical. It will be better for future styling to give them the same class. 

<img width="320" alt="Screenshot 2024-07-20 at 9 37 21 PM" src="https://github.com/user-attachments/assets/d489b61c-87da-43af-b6b2-8d958c1cbb3a">

annasvrv replied:
#fixed

arkhipenkav commented on category.html:
```diff
@@ -0,0 +1,499 @@
+          <button class="close-btn">&times;</button>
+        </div>
+
+        <form action="submit">
```
Nice try! You are not afraid of challenges 👍 You will have a lesson about forms soon. Let's leave here a comment with TODO. Ex.` <!-- TODO check this form after a lesson about forms -->`

arkhipenkav commented on category.html:
```diff
@@ -0,0 +1,499 @@
+        </button>
+        <button class="sandwish-bar-btn">
+          <img src="/" alt="Menu button" />
+        </button>
```
```
<button class="view-mode-btn">
    <img src="/" alt="View Mode button" />
</button>
<button class="sandwish-bar-btn">
    <img src="/" alt="Menu button" />
</button>
```
These two buttons look the same. They have different images inside and different states. We can show it in classes and make it easier for styling later. So, please, use the same class for common styles and add an `active` class to show a different state of one of the buttons.

<img width="227" alt="Screenshot 2024-07-20 at 9 47 36 PM" src="https://github.com/user-attachments/assets/3fbef291-4089-4844-9e1b-a267697427bc">

annasvrv replied:
#fixed

arkhipenkav commented on index.html:
```diff
@@ -3,7 +3,478 @@
+
+      <!-- Categories section -->
+      <section class="categories">
+        <h2 style="display: none">CATEGORIES</h2>
```
It is not necessary to start a section with a title. You have it inside in Category description block, it's enough for validity.

annasvrv replied:
#fixed

arkhipenkav commented on index.html:
```diff
@@ -3,7 +3,478 @@
+          </p>
+        </article>
+
+        <article class="category-card">
```
As you see in a mockup, this item has a different background color. Even though the styling work will start later, you can add a unique class for this exception. Ex. `accent`

`<article class="category-card accent">`

annasvrv replied:
#fixed

arkhipenkav commented on index.html:
```diff
@@ -3,7 +3,478 @@
+
+      <!-- Awards section -->
+      <section class="awards">
+        <h2 style="display: none">AWARDS</h2>
```
Again, you have this heading below in this section, it is enough. Check everywhere please

annasvrv replied:
#fixed

arkhipenkav commented on index.html:
```diff
@@ -3,7 +3,478 @@
+      <!-- Awards section -->
+      <section class="awards">
+        <h2 style="display: none">AWARDS</h2>
+        <!-- <img src="/" alt="Background Award image" /> -->
```
It is correct, you do not need an `img` tag for a background picture. You can remove it.

annasvrv replied:
#fixed

arkhipenkav commented on index.html:
```diff
@@ -3,7 +3,478 @@
+
+      <!-- Testimonials section -->
+      <section class="testimonials">
+        <!-- <img src="/" alt="Background image"> -->
```
The same here, you can remove it

annasvrv replied:
#fixed

arkhipenkav commented on index.html:
```diff
@@ -3,7 +3,478 @@
+          <p></p>
+        </details>
+
+        <details>
```
A great example of usage `details`
Just a small remark. To make this accordion open by default as we have in a mockup, you need to add the Boolean attribute `open` to a `details` tag
`<details open>`

annasvrv replied:
Thank you for this tip.

#fixed

arkhipenkav commented on index.html:
```diff
@@ -3,7 +3,478 @@
+        <!-- Blog articles -->
+
+        <article class="blog-card">
+          <p class="label">TYPOGRAPHY</p>
```
You have the same situation in a header navigation. It is more natural to use a `span` for a label

annasvrv replied:
#fixed

arkhipenkav commented on product-details.html:
```diff
@@ -0,0 +1,395 @@
+      name="description"
+      content="Explore our new exclusive digital design assets and graphic collections today!"
+    />
+    <title>Broduct Details</title>
```
A small typo here ;)

annasvrv replied:
#fixed

arkhipenkav commented on product-details.html:
```diff
@@ -0,0 +1,395 @@
+
+        <div class="column">
+          <h2>Product Details</h2>
+          <dl>
```
Nice solution with a description list for key-value pairs!

arkhipenkav commented on category.html:
```diff
@@ -0,0 +1,499 @@
+        <h2 style="display: none">catalog</h2>
+
+        <!-- Product Item -->
+        <article class="product-item">
```
These product items look like a list with an `articles` as a part of list items. Consider restructuring for clarify and better semantics

annasvrv replied:
#fixed

arkhipenkav commented on category.html:
```diff
@@ -0,0 +1,499 @@
+      </section>
+
+      <!-- Pagination -->
+      <div class="pagination">
```
This should be a list, usually named breadcrumbs. Breadcrumbs improve navigation and indicate the user's location within the website.

annasvrv replied:
I made it as div with links, because I googled and in every article it was suggested like that... 

https://www.w3schools.com/howto/howto_css_pagination.asp

I will leave it as it is for now and ask Liza for clarification.

arkhipenkav replied:
We've discussed this with Liza :) This solution that you've found is common but pretty old. For now, we have a better and more semantic solution such as a list with links. But, of course, you can ask for a deeper explanation at class

annasvrv replied:
Ah, I have understood now. I just need to wrap it as a list with links. 

#fixed

arkhipenkav commented on index.html:
```diff
@@ -3,7 +3,478 @@
+        <h2 style="display: none">CATEGORIES</h2>
+
+        <!-- Category cards -->
+        <article class="category-card">
```
These category cards look like a list with an articles as a part of list items. Please, check, maybe you have other places in document that should be restructured

annasvrv replied:
#fixed

arkhipenkav reviewed (commented):
Overall, it is a very good work! It is just need a few improvements. 

What I liked:

- No textual content within divs
- Usage of such tricky ones as `details`, `dl`, `article`, `inputs`, etc
- Excellent that you added `alt` attributes for all images. This is important for accessibility.
- Base semantic tags were used perfectly well: `header`, `main`, `footer` - all correct!
- Choice of links and buttons for clickable elements
- The same markup for repeated blocks - very good and important for a future work to notice these blocks

I really appreciate your self research and work with forms. It is a huge part of knowledge and you will learn everything soon.

I leave a few comments in the code. The main points I want you to pay attention at:

- usage of span tag
- usage of `article` and `section` tags
- breadcrumbs for navigation

The rest are more suggestions and should make your future work with styling easier. 

Please remember to add changes to the footer and header on all pages. I look forward to the updated request.

annasvrv committed:
Fixed markup

Changed some classes, added forms, fixed list with articles, cleaned up.

annasvrv committed:
Fixed markup

Changed some classes, added forms, fixed list with articles, cleaned up.

arkhipenkav commented on category.html:
```diff
@@ -0,0 +1,678 @@
+            <div class="range-container">
+              <div class="range-min">
+                $
+                <input type="number" class="price-nmb" />
```
Great fix!

Could you please add a `<label>` for an input? It is not a good practice for accessibility to use inputs without labels. Screen reader will read out loud the label, when the user is focused on the element.

annasvrv replied:
#fixed

arkhipenkav commented on category.html:
```diff
@@ -0,0 +1,678 @@
+
+        <button type="reset" class="btn-clearall">Clear all filters</button>
+
+        <!-- <div class="range-input">
```
You can remove this commented code.

annasvrv replied:
#fixed 

I left some commented code just till your approval and was going to remove before merge. Sorry I forgot to mention it.

arkhipenkav commented on category.html:
```diff
@@ -0,0 +1,678 @@
+      <!-- Copyright and terms -->
+      <div class="footer-copyright-section">
+        <ul>
+          <li><a href="#">Privacy Policy</a></li>
```
Please, add `target="_blank"` for external links.

annasvrv replied:
#fixed 

But I have a small question. How to decide if its external link. It is clear about social networks etc. Can we consider all links that doesn't have anchor link or other page from markup to be external links? Hope i make sense.

```
<section class="footer-navigation">
          <h2 class="footer-nav-title">Account</h2>
          <ul class="footer-nav-list">
            <li><a href="./cart.html">Your Cart</a></li>
            <li><a href="./checkout.html">Check Out</a></li>
            <li><a href="#" target="_blank">Refund Policy</a></li>
            <li><a href="#" target="_blank">Company Policy</a></li>
            <!-- anchor link -->
            <li><a href="#">Send A Message</a></li>
            <!-- mail link -->
            <li><a href="#">Customer Support</a></li>
          </ul>

<!-- Copyright and terms -->
      <div class="footer-copyright-section">
        <ul>
          <li><a href="#" target="_blank">Privacy Policy</a></li>
          <li><a href="#" target="_blank">Terms and Condition</a></li>
          <li><a href="#" target="_blank">Site Map</a></li>
        </ul>
```

arkhipenkav commented on product-details.html:
```diff
@@ -0,0 +1,555 @@
+        </div>
+
+        <div class="product-info-box">
+          <!-- <form action="" method="post">
```
You should know that in the real world it is not a good practice to leave commented code in project :) If you need it, you can keep it for now. But avoid it in the future or at least add a comment why this code was commented.

annasvrv replied:
#fixed 

I should have left a comment why I kept commented code. I didn't delete it as I wanted to clarify stars rating. And Liza confirmed it should be just decorative style, not a form. I guess I have the same form in index page. I will remove form too.

arkhipenkav reviewed (commented):
Such a great job. You've fixed almost everything and added a lot of code after a lecture about forms. 
Only few comments were added :) I believe you can quickly fix it and I will approve this PR

annasvrv committed:
Fixed html markup.

Added target blank for external links.
Removed star rating form.
Added label to price range inputs.
Added list to pagination links.
Cleaned up commented pieces of code.

annasvrv committed:
cleanup
